### PR TITLE
Added support for ModelDefinitions.views field

### PR DESCRIPTION
**Pull request by pawartur**
## Description

This will unblock us to enable devs to mark some fields as referencing document controller their GraphQL SDL definitions of models.

The `views` field should be:
```
interface ModelDefinition {
  ...
  views?: ModelViewsDefinition
}
```
, where
```
type ModelViewsDefinition = Record<string, ModelViewDefinition>
```
and
```
ModelViewDefinition = { type: 'documentAccount' }
```

## How Has This Been Tested?

Describe the tests that you ran to verify your changes. Provide instructions for reproduction.

- [X] Added unit tests for the new `ViewsValidation` interface
- [X] Updated tests for `ModelStreamHandler`

## PR checklist

Before submitting this PR, please make sure:

- [x] I have tagged the relevant reviewers and interested parties
- [x] I have updated the READMEs of affected packages
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts [hunk 1]
```diff
@@ -94,6 +94,52 @@ const FINAL_CONTENT = {
...
 }
 
+const FINAL_CONTENT_WITH_ACCOUNT_DOCUMENT_VIEW: ModelDefinition = {
+  name: 'MyModel',
+  accountRelation: ModelAccountRelation.LIST,
+  schema: {
+    $schema: 'https://json-schema.org/draft/2020-12/schema',
+    type: 'object',
+    properties: {
+      owner: {
```
Comment (stbrody) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:104]: I thought these view properties wouldn't show up in the schema itself, but it just metadata for the client library to expose the stream metadata with a given name?  @PaulLeCam

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts [hunk 2]
```diff
@@ -290,6 +336,18 @@ describe('ModelHandler', () => {
     await checkSignedCommitMatchesExpectations(did, commit, expectedGenesis)
   })
 
+  it('supports view properties in genesis comment', async () => {
```
Comment (stbrody) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:339]:
```suggestion
  it('supports view properties in genesis commit', async () => {
```

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts (hunk 1 above)
Comment (pawartur) [reply packages/stream-model-handler/src/__tests__/model-handler.test.ts:104]: Yes! That's right, sorry for the mistake.

**Review by PaulLeCam — APPROVED** [review]
Looks great, thanks!
I'm not the most familiar with this codebase so better wait for approval from other reviewers, but it looks all good to me.

File: packages/stream-model-handler/src/__tests__/model-handler.test.ts [hunk 3]
```diff
@@ -94,6 +94,46 @@ const FINAL_CONTENT = {
   },
 }
 
+const FINAL_CONTENT_WITH_ACCOUNT_DOCUMENT_VIEW: ModelDefinition = {
```
Comment (ukstv) [inline packages/stream-model-handler/src/__tests__/model-handler.test.ts:97]: Based on what I see, you could do something like
```typescript
{
  ...FINAL_CONTENT,
  views: {
    owner: {
      type: 'documentAccount'
    }
  }
}
```

File: packages/stream-model-handler/src/model-handler.ts [hunk 4]
```diff
@@ -125,6 +128,9 @@ export class ModelHandler implements StreamHandler<Model> {
 
     if (state.content.schema !== undefined) {
```
Comment (ukstv) [inline packages/stream-model-handler/src/model-handler.ts:129]:
```suggestion
    if (state.content.schema) {
```

IMO this should work too, and is a bit more idiomatic.

File: packages/stream-model-handler/src/model-handler.ts [hunk 5]
```diff
@@ -125,6 +128,9 @@ export class ModelHandler implements StreamHandler<Model> {
...
       await this._schemaValidator.validateSchema(state.content.schema)
+      if (state.content.views !== undefined) {
```
Comment (ukstv) [inline packages/stream-model-handler/src/model-handler.ts:131]:
```suggestion
      if (state.content.views) {
```

File: packages/stream-model-handler/src/views-utils.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+const SUPPORTED_VIEW_TYPES = [
+  'documentAccount'
+]
+
+export class ViewsValidation {
+  public validateViews(
+    views: ModelViewsDefinition,
+    schema: JSONSchema.Object
+  ) {
```
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:12]:
```suggestion
  ): void {
```

File: packages/stream-model-handler/src/views-utils.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    Object.entries(views).map(([key, value]) => {
```
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:13]:
```suggestion
    Object.entries(views).forEach(([key, value]) => {
```

you do not really return anything here

File: packages/stream-model-handler/src/views-utils.ts [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+      if (!SUPPORTED_VIEW_TYPES.includes(value.type)) {
```
Comment (ukstv) [inline packages/stream-model-handler/src/views-utils.ts:14]: Suggested micro-optimisation. Replace `SUPPORTED_VIEW_TYPES` array with an object like
```
{
documentAccount: true
}
```

Then this line would be O(1): `SUPPORTED_VIEW_TYPES[value.type]`.

**Review by ukstv — APPROVED** [review]
few beautifications here and there are possible, but otherwise LGTM

🏇

**Review by stbrody — APPROVED** [review]
two small nitpicks, but overall looks good

**Pull request merged**